File: GMS-backend/GMS/Entities/Endpoint/Korisnik_Suplement/Edit/Korisnik_SuplementEditEndpoint.cs
using GMS.Data;
using GMS.Helpers;
using GMS.Helpers.Auth;
using Microsoft.AspNetCore.Mvc;

namespace GMS.Entities.Endpoint.Korisnik_Suplement.Edit
{
    [Route("Korisnik_Suplement-Edit")]
    [MyAuthorization]


    public class Korisnik_SuplementEditEndpoint : MyBaseEndpoint<Korisnik_SuplementEditRequest, int>
    {
        private readonly ApplicationDbContext db;

        public Korisnik_SuplementEditEndpoint(ApplicationDbContext db)
        {
            this.db = db;
        }

        [HttpPost]
        public override async Task<int> Handle([FromBody]Korisnik_SuplementEditRequest request, CancellationToken cancellationToken)
        {
            Models.Korisnik_Suplement? korisnik_suplement;
            if (request.KorisnikID == 0)
            {
                korisnik_suplement = new Models.Korisnik_Suplement();
                db.Add(korisnik_suplement);


            }
            else
            {
                korisnik_suplement = db.Korisnik_Suplement.FirstOrDefault(s => s.KorisnikID == request.KorisnikID && s.SuplementID == request.SuplementID && s.DatumVrijemeNarudzbe == request.DatumVrijemeNarudzbe);
                if (korisnik_suplement == null)
                    throw new Exception("pogresan ID");
            }

 
            korisnik_suplement.Isporuceno = request.Isporuceno;

           
            


            await db.SaveChangesAsync(cancellationToken);

            return korisnik_suplement.KorisnikID;
        }
    }
}

File: GMS-frontend/src/app/endpoints/korisnik-endpoints/korisnik-suplementi-endpoints/korisnik-suplement-edit-endpoint.ts

import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MyBaseEndpoint} from "../../MyBaseEndpoint";
import {Config} from "../../../config";
@Injectable({providedIn: 'root'})
export class Korisnik_SuplementEditEndpoint implements  MyBaseEndpoint<Korisnik_SuplementEditRequest, number>{
  constructor(public httpClient:HttpClient) { }

  Handle(request: Korisnik_SuplementEditRequest): Observable<number> {
    let url= Config.adresa + `Korisnik_Suplement-Edit`;

    return this.httpClient.post<number>(url, request);
  }

}
export interface Korisnik_SuplementEditRequest {
  korisnikID: number
  suplementID: number
  datumVrijemeNarudzbe: string
  kolicina: number
  isporuceno: boolean
}

